Sources/Sentry/SentryRequest.m

```diff
@@ -0,0 +1,51 @@
+#import "SentryRequest.h"
+#import "NSDictionary+SentrySanitize.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentryRequest
+
+- (instancetype)init
+{
+    return [super init];
+}
+
+- (NSDictionary<NSString *, id> *)serialize
+{
+    NSMutableDictionary *serializedData = [[NSMutableDictionary alloc] init];
+
+    @synchronized(self) {
+        [serializedData setValue:self.bodySize forKey:@"body_size"];
+        [serializedData setValue:self.cookies forKey:@"cookies"];
+        [serializedData setValue:self.fragment forKey:@"fragment"];
+        [serializedData setValue:[self.headers sentry_sanitize] forKey:@"headers"];
+        [serializedData setValue:self.method forKey:@"method"];
+        [serializedData setValue:self.queryString forKey:@"query_string"];
+        [serializedData setValue:self.url forKey:@"url"];
```

Comment:
I noticed that this method never checks for nullability, what if `headers` is null, won't it throw?

Reply:
objc dont send messages to null objects, and return null as result.
Dictionaries accept null as value to remove the key.

Reply:
ok so I assume I don't need any null checks here, all good.

Reply:
No, it won't, but it will add the headers with an empty value. You can check for `nil` to make the payload smaller.